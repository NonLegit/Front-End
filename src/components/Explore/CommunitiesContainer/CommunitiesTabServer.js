import { useState, useEffect } from 'react';
import axios from '../../../services/instance';

/**
*
* Function to perfom api request and fetch communities
* @param {string} name - title of explore page
* @returns {Array} array of communities
*/

export const exploreCommunities = (name) => {
  const [data, setData] = useState([]);

  const api = `/t/${name}/`;
  useEffect(() => {
    axios.get(api) // fetch api
      .then((actualData) => {
        console.log(actualData.data);
        setData(actualData.data);
      })
      .catch((error) => {
        if (error.response.status === 401) {
          window.location.href = './login';
        }

        console.log(error);
      });
  }, [api]);
  return [data];
};
