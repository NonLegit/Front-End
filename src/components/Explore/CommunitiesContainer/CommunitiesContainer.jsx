import { useEffect } from 'react';
import { useHeaderSubtitleContext } from '../../../contexts/HeaderSubtitleContext';
import CommunitiesHeader from '../CommunitiesHeader/CommunitiesHeader';
import Community from '../Community/Community';
import { StyledCommunitiesContainer } from './styles';
import { exploreCommunities } from './CommunitiesTabServer';

/**
 * communities and their header
 * @component
 * @property {string} title title of the communities
 * @return {React.Component} - communities and their header
 */

function CommunitiesContainer(props) {
  const { title } = props;
  const [communities] = exploreCommunities(title);
  const { setHeaderSubtitle } = useHeaderSubtitleContext();
  useEffect(() => {
    setHeaderSubtitle(communities?.type);
    console.log('fady', communities?.type);
  }, [communities]);
  return (
    <StyledCommunitiesContainer>
      <CommunitiesHeader explore={title} />
      {
      communities?.data?.map((commuity) => {
        console.log(commuity);
        const {
          _id, name, picture, membersCount, description,
        } = commuity;
        console.log(_id);
        return (
          <Community
            communityName={name}
            communityPicture={picture}
            numOfMembers={membersCount}
            description={description}
          />
        );
      })
    }
    </StyledCommunitiesContainer>
  );
}

export default CommunitiesContainer;
