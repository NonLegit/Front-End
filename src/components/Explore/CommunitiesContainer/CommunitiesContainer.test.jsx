import renderer from 'react-test-renderer';
import CommunitiesContainer from './CommunitiesContainer';
import HeaderSubtitleContextProvider from '../../../contexts/HeaderSubtitleContext';

test('test snapshot', async () => {
  const tree = renderer.create(
    <HeaderSubtitleContextProvider>
      <CommunitiesContainer title="sports" />
    </HeaderSubtitleContextProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
