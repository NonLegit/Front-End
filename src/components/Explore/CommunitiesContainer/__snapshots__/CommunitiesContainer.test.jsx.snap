// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-1d1c013"
>
  <div
    className="MuiBox-root css-12205oe"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Communities related to
       
      sports
    </p>
  </div>
</div>
`;
